js/page-specific/baltcogo-app/BaltCoGoReporterCtrl.js

```diff
@@ -1,504 +1,530 @@
-
-			if (!$target.is('.autocomplete-results button')) {
-				return;
-			}
-			
-			if (keycode === 27 && angular.element('.autocomplete-results').is(':visible')) {
-				self.autocompleteResults = [];
-				$scope.$apply();
-			}
-
-			event.preventDefault();			
-
-			if (keycode === 13 || keycode === 32) {
-				$target.trigger('click');
-			}
-
-			if (keycode === 40) {
-				$target.parent().next().find('button').focus();
-			}
-
-			if (keycode === 38) {
-				$target.parent().prev().find('button').focus();
-			}
-		}
-
-		function animalTypeSuccessHandler(response) {
-			self.animalTypeData = response.data;
-		}
-
-		function breedSuccessHandler(response) {
-			self.animalBreedData = response.data;
-		}
-
-		function categorySuccessHandler(response) {
-			self.categoryData = response.data;
-			if (categoryId)
-				autoSelectCategories(categoryId);
-		}
-
-		function colorSuccessHandler(response) {
-			self.animalColorData = response.data;
-		}
-
-		function displayAddressError() {
-			angular.element('#map').closest('cs-form-control').addClass('error');
-			$scope.citySourcedReporterForm.address.$setDirty();
-			self.address = '';
-			$scope.$apply();
-		}
-
-		function errorHandler(err) {
-			console.log(err);
-		}
-
-		function mapClickHandler(event) {			
-			var $wrapper = angular.element('#map').closest('cs-form-control'),
-				addressField = $scope.citySourcedReporterForm.address;
-			
-			self.autocompleteResults = [];
-			self.latitude = event.latLng.lat();
-			self.longitude = event.latLng.lng();
-
-			mapServiceComposite.reverseGeocode(self.latitude, self.longitude, function (response) {
-				$wrapper.removeClass('error');
-				mapServiceComposite.createMarker(map, self.latitude, self.longitude);
-				self.address = response.address.Street.toLowerCase() + ', ' + response.address.City.toLowerCase() + ', ' + response.address.State.toUpperCase();
-				$scope.$apply();
-			}, function(err) {
-				$wrapper.addClass('error');
-				addressField.$setDirty();
-				self.address = '';				
-				$scope.$apply();
-			});
-		}
-
-		function petTypeSuccessHandler(response) {
-			self.petTypeData = response.data;
-		}
-
-		function preventSubmitOnEnterPressHandler(event) {
-			var keyCode = event.which || event.keyCode;
-
-			if (keyCode === 13) {
-				event.preventDefault();
-			}		
-		}
-
-	}
+  'use strict';
+
+  function Data(arr) {
+    this.endpoints = {};
+
+    for (var i = 0; i < arr.length; i++) {
+      var obj = arr[i];
+      console.log(obj);
+      var key = Object.keys(obj)[0];
+
+      this.endpoints[key] = obj[key];
+    }
+  }
+
+  app.provider('data', function () {
+    var urlArr = null;
+
+    this.setUrls = function (keyValArr) {
+      urlArr = keyValArr;
+    };
+
+    this.$get = [function () {
+      return new Data(urlArr);
+    }];
+  });
+
+
+  app.controller('BaltCoGoReporterCtrl', ['$http', '$scope', '$timeout', 'mapServiceComposite', 'reportService', "data", reporterController]);
```

Comment:
"data" should be in single quotes, but see my comment below about giving it a better name.